blog/2024-11-mastering-effect-and-afterRenderEffect/README.md

```diff
@@ -0,0 +1,477 @@
+
+For patching forms, there is currently no better approach than using effects. 
+This approach can also be easily migrated to what would have been done in the past with `ngOnchanges` – which is great.
+But whether we should have used a better computed signal for the `isEditMode` is questionable.
+It is not easy to make a decision here, because we could also have written the following:
+
+```ts
+isEditMode = computed(() => !!this.currentBook());
+```
+
+Perhaps we have to accept that in some situations both options are absolutely valid!
+
+
+## Introducing `afterRenderEffect()`
+
+The new `afterRenderEffect()` function allows us to control when tasks occur during the DOM update process.
+This is particularly beneficial for UI manipulations that require specific timing to avoid layout shifts and ensure smooth animations.
+
+The API itself mirrors the functionality of 
+* [`afterRender`](https://next.angular.dev/api/core/afterRender) *(registers a callback to be invoked each time the application finishes rendering)* and 
+* [`afterNextRender`](https://next.angular.dev/api/core/afterNextRender) *(Registers a callbacks to be invoked the next time the application finishes rendering, during the specified phases.)* 
+
+which are both in **Developer Preview**!
+
+The Angular docs recommend avoiding `afterRender` when possible and suggest specifying explicit phases with `afterNextRender` to avoid significant performance degradation. 
+You’ll see a similar recommendation for `afterRenderEffect()`. There is one signature that is intended for use and another that exists but is not recommended.
+
+But there is one big difference:
+> **💡 Values are propagated from phase to phase as signals instead of as plain values.** 
+
+As a result, later phases may not need to execute if the values returned by earlier phases do not change - and if there is no other dependency established (we will talk about this soon). Before we start, some important facts to know about the effects created by `afterRenderEffect()`:
+
+* **Phased Execution**: These effects can be registered for specific phases of the render cycle. 
+  The Angular team recommends adhering to these phases for optimal performance.
+* **Signal Integration**: These effects are supposed to work seamlessly with Angular’s signal reactivity system, and signals can be set during the phases.
+* **Selective Execution**: These effects only rerun when they are "dirty" due to signal dependencies. If no signal changes, the effect won’t trigger again.
+* **No SSR**: These effects execute only in browser environments, not on the server.
+
+
+### Understanding the Phases
+
+Phased execution is useful for avoiding unnecessary layout recalculations.
+We can register for each phase by specifying a callback function.
+The first callback receive no parameters.
+Each subsequent phase callback will receive the return value of the previously phase **as a signal**.
+This can be used to coordinate work across multiple phases.
+
```

Comment:
Before you dig into the phases, I struggle to understand when the effect will run. Will it execute whenever anything in the DOM changes, e.g. through interpolation or property bindings? If yes, write this clearly. I'm reading a lot of details here, but I haven't even learned the basics.